app/assets/javascripts/admin/controllers/modals/admin-install-theme.js.es6

```diff
@@ -0,0 +1,278 @@
+  create: Ember.computed.equal("selection", "create"),
+  selection: "popular",
+  adminCustomizeThemes: Ember.inject.controller(),
+  loading: false,
+  keyGenUrl: "/admin/themes/generate_key_pair",
+  importUrl: "/admin/themes/import",
+  checkPrivate: Ember.computed.match("uploadUrl", /^git/),
+  localFile: null,
+  uploadUrl: null,
+  urlPlaceholder: "https://github.com/discourse/sample_theme",
+  advancedVisible: false,
+  themesController: Ember.inject.controller("adminCustomizeThemes"),
+  createTypes: [
+    { name: I18n.t("admin.customize.theme.theme"), value: THEMES },
+    { name: I18n.t("admin.customize.theme.component"), value: COMPONENTS }
+  ],
+
+  @computed("themesController.installedThemes")
+  themes(installedThemes) {
+    return POPULAR_THEMES.map(t => {
+      if (installedThemes.includes(t.name)) {
+        Ember.set(t, "installed", true);
+      }
+      return t;
+    });
+  },
+
+  @computed("themesController.installedThemes")
+  popularComponents(installedThemes) {
+    return POPULAR_COMPONENTS.map(t => {
+      if (installedThemes.includes(t.name)) {
+        Ember.set(t, "installed", true);
+      }
+      return t;
+    });
+  },
+
+  @computed(
+    "loading",
+    "remote",
+    "uploadUrl",
+    "local",
+    "localFile",
+    "create",
+    "nameTooShort"
+  )
+  installDisabled(
+    isLoading,
+    isRemote,
+    uploadUrl,
+    isLocal,
+    localFile,
+    isCreate,
+    nameTooShort
+  ) {
+    return (
+      isLoading ||
+      (isRemote && !uploadUrl) ||
+      (isLocal && !localFile) ||
+      (isCreate && nameTooShort)
+    );
+  },
+
+  @observes("privateChecked")
+  privateWasChecked() {
+    this.get("privateChecked")
+      ? this.set("urlPlaceholder", "git@github.com:discourse/sample_theme.git")
+      : this.set("urlPlaceholder", "https://github.com/discourse/sample_theme");
+
+    const checked = this.get("privateChecked");
+    if (checked && !this._keyLoading) {
+      this._keyLoading = true;
+      ajax(this.get("keyGenUrl"), { method: "POST" })
+        .then(pair => {
+          this.set("privateKey", pair.private_key);
+          this.set("publicKey", pair.public_key);
+        })
+        .catch(popupAjaxError)
+        .finally(() => {
+          this._keyLoading = false;
+        });
+    }
+  },
+
+  @computed("name")
+  nameTooShort(name) {
+    return !name || name.length < MIN_NAME_LENGTH;
+  },
+
+  @computed("component")
+  placeholder(component) {
+    if (component) {
+      return I18n.t("admin.customize.theme.component_name");
+    } else {
+      return I18n.t("admin.customize.theme.theme_name");
+    }
+  },
+
+  @computed("themesController.currentTab")
+  selectedType(tab) {
+    return tab;
+  },
+
+  @computed("selectedType")
+  component(type) {
+    return type === COMPONENTS;
+  },
+
+  @computed("selection")
+  submitLabel(selection) {
+    if (selection === "create") return "admin.customize.theme.create";
+    else return "admin.customize.theme.install";
```

Comment:
trivial: The `else` here is not required

Reply:
Or write is as 

``` javascript
return `admin.customize.theme.${selection === "create" ? "create" : "instal"}`;